utils/upodman/Makefile

```diff
@@ -0,0 +1,106 @@
+  controlling and statistics over pods and containers through
+  ubus.
+
+  upodman is written in C++ except for ubus parts which are
+  externed from C. It uses staticly linked libcurl for unix
+  socket operations, because system's default libcurl might
+  not support unix sockets.
+endef
+
+MAKE_VARS += \
+	BUILD_DIR="$(PKG_BUILD_DIR)"
+
+CURL_VERSION:=7.87.0
+CURL_PATH:=$(PKG_BUILD_DIR)/curl-$(CURL_VERSION)
+CONFIGURE_PATH:=curl-$(CURL_VERSION)
+
+CONFIGURE_ARGS+= \
+		--disable-debug --enable-optimize --disable-curldebug --disable-ares \
+		--enable-rt --disable-ech --enable-shared=no --enable-static=yes --enable-http \
+		--disable-ftp --disable-file --disable-ldap --disable-ldaps --disable-rtsp \
+		--disable-proxy --disable-dict --disable-telnet --disable-tftp --disable-pop3 \
+		--disable-imap --disable-smb --disable-smtp --disable-gopher --disable-mqtt \
+		--disable-manual --disable-ipv6 --enable-pthreads --enable-verbose --disable-sspi \
+		--disable-crypto-auth --disable-ntlm --disable-ntlm-wb --disable-tls-srp \
+		--enable-unix-sockets --disable-cookies --disable-socketpair --disable-http-auth \
+		--disable-doh --disable-mime --disable-dateparse --disable-netrc --disable-progress-meter \
+		--disable-dnsshuffle --enable-get-easy-options --disable-alt-svc --disable-headers-api \
+		--disable-hsts --disable-websockets --without-schannel --without-secure-transport \
+		--without-amissl --without-ssl --without-zlib --without-brotli --without-zstd \
+		--without-gssapi --without-ca-bundle --without-libpsl --without-libgsasl \
+		--without-libssh2 --without-libssh --without-wolfssh --without-librtmp \
+		--without-winidn --without-libidn2 --without-nghttp2 --without-ngtcp2 \
+		--without-nghttp3 --without-quiche --without-msh3 \
+		--without-zsh-functions-dir --without-fish-functions-dir
+
+define Build/Configure
+endef
+
+define Build/Prepare
+	$(call Build/Prepare/Default)
+	mkdir -p $(CURL_PATH)
+	$(TAR) xzf $(PKG_BUILD_DIR)/dl/curl-$(CURL_VERSION).tar.gz -C $(PKG_BUILD_DIR)
+endef
+
+define Build/Configure
+	$(call Build/Configure/Default)
+	$(SED) 's#DIST_SUBDIRS = \$$(SUBDIRS) tests packages scripts include docs#DIST_SUBDIRS = \$$(SUBDIRS) packages scripts include#g' $(CURL_PATH)/Makefile
+	$(SED) 's#cd docs && \$$(MAKE) install#echo not-installing-docs#g' $(CURL_PATH)/Makefile
+	$(SED) 's#cd docs/libcurl && \$$(MAKE) install#echo not-installing-docs#g' $(CURL_PATH)/Makefile
+	$(SED) 's#SUBDIRS = ../docs#SUBDIRS =#g' $(CURL_PATH)/src/Makefile
+	$(SED) 's#SUBDIRS = ../docs#SUBDIRS =#g' $(CURL_PATH)/src/Makefile.am
+endef
+
+define Build/Compile/Curl
+	+$(MAKE_VARS) \
+	$(MAKE) $(PKG_JOBS) -C $(CURL_PATH) \
+		$(MAKE_FLAGS);
+endef
+
+define Build/Install/Curl
+	$(MAKE_VARS) \
+	$(MAKE) -C $(CURL_PATH) \
+		$(MAKE_FLAGS) \
+	DESTDIR="$(CURL_PATH)/inst" \
+	install ;
+endef
+
+define Build/Compile
+	$(call Build/Compile/Curl)
+	$(call Build/Install/Curl)
+	$(call Build/Compile/Default)
+endef
```

Comment:
No, no no, just no. Most likely, you are looking for HostBuild of curl. Why are you using curl at all?

Reply:
No no no and just no. Host curl? Why on earth? You've got it all wrong. Original makefile builds curl, but unfortunately all flags don't seem to carry-on to makefile that is included with include directive, that's why I added built-in curl building to package Makefile. I need target built curl library, either dynamic or static, but it must have support for unix sockets.

Reply:
Yup, I got it wrong, but that's caused that I don't know your package and it was not properly explained, why you put curl in your's upodman repository. 

If I understand it correctly, you want to add dependency to your's upodman package on [LIBCURL_UNIX_SOCKETS](https://github.com/openwrt/packages/blob/a79df49a5fb7fed5a525c72824c966261cbef62e/net/curl/Config.in#L140) instead of just trying to hack it in very bad way.

Reply:
Also, you need to take it from the security point of view. We can not trust your sources that you put some curl tarball in your repository if you don't verify the checksum, it could be corrupted, your account could be hijacked, the tarball can be changed and we are screw up. Don't we?

Reply:
Many software included "built-in" software, it's as likely as any software being hijacked or whatever, why would someone instead attack included curl source, when they can put their code in very much simpler set of mine? It's verified **in** sha256sum of package as it's pre-downloaded, so that more or less breaks that assumption; go ahead, download source code and then download identical version from curl's site and verify they are the same, then you know that sha256sum of release covers that, if you need verification.

If I put dependency on LIBCURL_UNIX_SOCKETS which is not enabled by default, how can this package ever be available publicly with opkg, as I was asked to deliver it as? I ain't gonna start compiling it for anything else except x86-64...

Reply:
> go ahead, download source code and then download identical version from curl's site and verify they are the same, then you know that sha256sum of release covers that, if you need verification.

C'mon, imagine my situation, when I clone all the source code from OpenWrt, compile all the packages and I should verify manually each tarball, which was donwloaded as it is build-in? You can not be serious about that.  That's what the build system should do and it does, but that's not in your case. We are trusting you and anything might happen.

> If I put dependency on LIBCURL_UNIX_SOCKETS which is not enabled by default, how can this package ever be available publicly with opkg, as I was asked to deliver it as?

How it is done for other packages? 🤔 Did you check any other packages before arguing about that?

Reply:
It's done the same way, **except** source code is pre-extracted. This way release tar ball takes less space, and it's also possible to verify the curl source. Though, I only designed it in the first place so that it would be a smaller download.